clients/src/main/java/org/apache/kafka/common/utils/Bytes.java

```diff
@@ -139,7 +140,7 @@ private static String toString(final byte[] b, int off, int len) {
      */
     public final static Comparator<byte[]> BYTES_LEXICO_COMPARATOR = new LexicographicByteArrayComparator();
 
-    private interface ByteArrayComparator extends Comparator<byte[]> {
+    private interface ByteArrayComparator extends Comparator<byte[]>, Serializable {
```

Comment:
The JavaDoc for java.util.Comparator says that:
```
Note: It is generally a good idea for comparators to also implement java.io.Serializable, as they may be used as ordering methods in serializable data structures (like TreeSet, TreeMap). In order for the data structure to serialize successfully, the comparator (if provided) must implement Serializable.
```
(see http://docs.oracle.com/javase/7/docs/api/java/util/Comparator.html)

There seems to be no harm in implementing Serializable here since the Comparator has no internal state, so there's nothing to do.